dace/codegen/targets/framecode.py

```diff
@@ -352,6 +380,10 @@ def generate_footer(self, sdfg: SDFG, global_stream: CodeIOStream, callsite_stre
                 callsite_stream.write("}")
 
         callsite_stream.write('delete __state;\n', sdfg)
+        if sdfg.instrument is dtypes.InstrumentationType.GPU_TX_MARKERS:
+            gpu_tx_markers_provider = self._dispatcher.instrumentation.get(dtypes.InstrumentationType.GPU_TX_MARKERS)
+            if gpu_tx_markers_provider:
+                gpu_tx_markers_provider.print_range_pop(callsite_stream)
```

Comment:
if you want to be able to instrument init/exit codes, you should add another `on_sdfg_{init,exit}_{begin,end}()` event API calls to `InstrumentationProvider`. Otherwise this is a special case that looks odd.